referenceqvm/tests/test_unitary_generator.py

```diff
@@ -208,3 +210,36 @@ def test_tensor_gates_two_qubit():
     test_unitary = tensor_gates(gate_matrix, {}, prog.actions[0][1], 4).toarray()
     true_unitary = apply_gate(gate_matrix['CNOT'], [1, 3], 4).toarray()
     assert np.allclose(test_unitary, true_unitary)
+
+
+def test_tensor_up():
+    """Testing tesnor up type checking and correctness"""
+    x_term = PauliTerm("X", 5)
+
+    # testing type rejection
+    with pytest.raises(TypeError):
+        tensor_up(x_term, 5)
+
+    # testing index rejection
+    with pytest.raises(IndexError):
+        tensor_up(PauliSum([x_term]), 3)
+
+    xy_term = PauliSum([PauliTerm("X", 0)*PauliTerm("Y", 1)])
+
+    # test correctness
+    trial_matrix = tensor_up(xy_term, 2)
+    true_matrix = np.kron(gate_matrix['Y'], gate_matrix['X'])
+    assert np.allclose(trial_matrix, true_matrix)
```

Comment:
don't use `np.allclose` with asserts. There is better functionality in numpy. Try
```
np.testing.assert_allclose(trial_matrix, true_matrix)
```
ass this not only raises assertion error but also tells you more details of what is wrong.